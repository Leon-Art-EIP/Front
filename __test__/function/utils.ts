import type { ReactTestRendererJSON } from "react-test-renderer";
// eslint-disable-next-line import/no-extraneous-dependencies
import renderer from "react-test-renderer";
import { expect } from "vitest";

function toJson(component: renderer.ReactTestRenderer): ReactTestRendererJSON {
  const result = component.toJSON();
  expect(result).toBeDefined();

  return result as ReactTestRendererJSON;
}

export function testSnapshot(element: JSX.Element): void {
  const component = renderer.create(element);
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
}
