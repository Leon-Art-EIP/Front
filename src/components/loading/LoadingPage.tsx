import { CircularProgress } from "@mui/material";

export default function LoadingPage() {
  return (
    <div className="flex items-center justify-center h-screen w-screen text-tertiary">
      <div className="text-center">
        <h1 className="text-xl font-bold mb-4">Chargement</h1>
        <CircularProgress size={50} thickness={4} color="primary" />
      </div>
    </div>
  );
}
