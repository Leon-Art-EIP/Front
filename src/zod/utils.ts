export const getValueOrUndefined = (arg: unknown) =>
  arg === "" || arg === undefined || arg === null ? undefined : arg;
