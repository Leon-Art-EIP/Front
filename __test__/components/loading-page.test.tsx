import { test } from "vitest";
import { testSnapshot } from "../function/utils";
import LoadingPage from "../../src/components/loading/LoadingPage";

test("LoadingPage snapshot", () => {
  testSnapshot(<LoadingPage />);
});
