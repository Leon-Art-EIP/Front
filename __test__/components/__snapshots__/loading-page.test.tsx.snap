// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LoadingPage snapshot 1`] = `
<div
  className="flex items-center justify-center h-screen w-screen text-tertiary"
>
  <div
    className="text-center"
  >
    <h1
      className="text-xl font-bold mb-4"
    >
      Chargement
    </h1>
    <span
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate MuiCircularProgress-colorPrimary css-18lrjg1-MuiCircularProgress-root"
      role="progressbar"
      style={
        {
          "height": 50,
          "width": 50,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg css-1idz92c-MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate css-176wh8e-MuiCircularProgress-circle"
          cx={44}
          cy={44}
          fill="none"
          r={20}
          strokeWidth={4}
          style={{}}
        />
      </svg>
    </span>
  </div>
</div>
`;
